File: JS/almost-palindrome.js
function isPalindromeRange(str, left, right) {
    while (left < right) {
        if (str[left] !== str[right]) return false;
        left++;
        right--;
    }
    return true;
}

function almostPalindrome(str) {
    let left = 0;
    let right = str.length - 1;

    while (left < right) {
        if (str[left] !== str[right]) {
            return isPalindromeRange(str, left + 1, right) ||
                isPalindromeRange(str, left, right - 1);
        }
        left++;
        right--;
    }

    return true;
}

// --- Read input from stdin ---
const input = require('fs').readFileSync('/dev/stdin', 'utf8').trim();
console.log(almostPalindrome(input) ? 'YES' : 'NO');

File: C++/almost-palindrome.cpp
#include <bits/stdc++.h>
using namespace std;

bool isPalindromeRange(const string &s, int left, int right) {
    while (left < right) {
        if (s[left] != s[right]) return false;
        left++;
        right--;
    }
    return true;
}

bool almostPalindrome(const string &s) {
    int left = 0, right = (int)s.size() - 1;

    while (left < right) {
        if (s[left] != s[right]) {
            return isPalindromeRange(s, left + 1, right) ||
                   isPalindromeRange(s, left, right - 1);
        }
        left++;
        right--;
    }

    return true;
}

int main() {
    string s;
    getline(cin, s);
    cout << (almostPalindrome(s) ? "YES" : "NO") << endl;
    return 0;
}